File: scripts/reference-solutions/meeting-rooms.cpp
#include <algorithm>
#include <iostream>
#include <utility>
#include <vector>

using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  int n;
  if (!(cin >> n)) {
    cout << "true\n";
    return 0;
  }

  vector<pair<long long, long long>> intervals(n);
  for (auto& interval : intervals) {
    cin >> interval.first >> interval.second;
  }

  sort(intervals.begin(), intervals.end());
  bool hasPrevious = false;
  long long previousEnd = 0;

  for (const auto& [start, end] : intervals) {
    if (hasPrevious && start < previousEnd) {
      cout << "false\n";
      return 0;
    }
    previousEnd = hasPrevious ? max(previousEnd, end) : end;
    hasPrevious = true;
  }

  cout << "true\n";
  return 0;
}

File: scripts/reference-solutions/meeting-rooms.py
import sys

data = sys.stdin.read().strip().split()
if not data:
    print("true")
    sys.exit(0)

n = int(data[0])
intervals = []
index = 1
for _ in range(n):
    start = int(data[index])
    end = int(data[index + 1])
    index += 2
    intervals.append((start, end))

intervals.sort()
previous_end = None
for start, end in intervals:
    if previous_end is not None and start < previous_end:
        print("false")
        break
    previous_end = max(previous_end, end) if previous_end is not None else end
else:
    print("true")
